core/spec/models/spree/product_spec.rb

```diff
@@ -14,6 +14,38 @@ class Extension < Spree::Base
     let(:product) { create(:product) }
     let(:variant) { create(:variant, product: product) }
 
+    describe '#cache_key' do
+      context 'when price changes in a week' do
+        let!(:current_price) { variant.prices.first }
+        let!(:future_price) { create(:price, variant: variant, valid_from: 1.week.from_now) }
+
+        context 'tomorrow' do
+          it 'the cache key does not change' do
+            old_cache_key = product.cache_key
+            allow(Time).to receive(:current).and_return(1.day.from_now)
+            expect(product.cache_key).to eq(old_cache_key)
+          end
+
+          context 'when updating the price' do
+            it 'changes the cache key' do
+              old_cache_key = product.cache_key
+              allow(Time).to receive(:current).and_return(1.day.from_now)
+              current_price.update(amount: current_price.amount + 1)
+              expect(product.cache_key).to_not eq(old_cache_key)
```

Comment:
I think that this test might be best expressed using an assertion like:

```Ruby
expect { subject }.to chance { product.cache_key }
```

Since that's what's important here.